### PR TITLE
fix(proxy): enforce an empty domain allowlist instead of allowing all

### PR DESCRIPTION
**Pull request by Starefossen**
An **enabled** domain allowlist that resolved to zero entries meant allow-all, because enforcement gated on `!allowed_domains.is_empty()` and could not tell "no allowlist configured" from "an allowlist with nothing on it". A configured-but-missing file fell through the same way, and startup only printed `(not found)`, which `-q` suppressed. `docs/proxy.md` promised the opposite: "Only listed domains can connect."

Four states, two of them wrong before this change:

| State | Before | After |
|---|---|---|
| Not configured | allow-all | allow-all (unchanged, intended) |
| File configured, empty or all comments | **allow-all** | denies every domain |
| File configured, missing at startup | **allow-all**, message suppressible | refuses to start |
| File configured, populated | enforced | enforced |

## The fix

`NetPolicy` gains `allowlist_active`, computed in `DomainPolicy::build` from the *sources* (`default_allowlist`, `allowed_domains_file`, `allowed_domains_initial`) before any of them is read, so a list that parses to zero entries still counts as active. The gate becomes `(allowlist_active || !allowed_domains.is_empty())` — the second half kept deliberately as a fail-closed backstop, so a caller that forgets to set the bit still enforces the list it built rather than silently allowing everything.

`cplt check net` sets the same bit. That half matters as much as the enforcement change: `check` previously reused the same `is_empty()` semantics, so a user verifying their setup got a self-consistent wrong answer. There is now a test asserting check's verdict equals the live policy's verdict host by host.

`ProxyState::net_policy` was inlining the `NetPolicy` literal; it now delegates to `DomainPolicy::net_policy(now)`, which is what lets tests exercise the real snapshot instead of a hand-rolled copy, and removes the place the two could drift.

## Breaking change, deliberately

Anyone shipping `proxy.allowed_domains` with the file empty or absent on some machines is running allow-all today without knowing it. They now get a hard failure instead. CI images that create the file lazily will break. That is the point — the alternative is leaving them believing they have egress control they do not have.

The refusal names both ways to mean "no allowlist", because the distinction is the whole finding:

```
[cplt] Domain allowlist file not found: /path/to/nope.txt
  This run was configured to restrict egress to an allowlist, so cplt refuses to start rather than come up allowing every domain.
  Fix it in one of these ways:
    - create /path/to/nope.txt with the domains you want reachable, one per line;
    - drop the `allowed_domains` key (or --allowed-domains) to run with no allowlist, which allows all domains;
    - pass --allow-all-domains to allow every domain for this run only.
  Note that an EMPTY allowlist file is not the same as no allowlist: it is an allowlist with nothing on it, and blocks every domain.
```

`--allow-all-domains` already existed and already covers the intentional case; nothing new was needed.

## What is deliberately unchanged

The hot-reload asymmetry: an **emptied** file tightens (it is a real edit), while a **deleted or unreadable** file keeps last-good, which is what `docs/proxy.md:150` promises. Both halves are tested.

`default_allowlist` composition: with `proxy.default_allowlist` on, `current()` returns `sticky ∪ file`, so emptying the file revokes only the user's own additions and the agent's defaults survive. The exposure was specifically an explicit `allowed_domains` file *without* `--default-allowlist`.

## Review notes

Reviewed independently before merge. Verified there: the bit survives hot reload (it is never written after `build`); `--allow-all-domains` yields false on both the proxy and check sides because the `FeatureToggle` merge forces `default_allowlist` off and `use_allowed_file` nulls the path; both refusal sites are reached, print in full under `-q`, and exit 1; `--print-profile` returns before the proxy and `--doctor` never builds a policy, so neither reaches allow-all; and a repo `.cplt.toml` cannot propose `allowed_domains`, so no untrusted path can trip the new refusal.

One premise from the original report did **not** survive: the built-in allowlist is not empty for `pi`, `goose` or `shell`. `default_allowed_domains()` appends `PACKAGE_REGISTRY_DOMAINS` for every agent; only the per-agent infrastructure slice is empty for those three. (A doc in another repo repeated that error and is being corrected separately.)

A replaced test, `allowlist_falls_back_to_initial_until_the_file_exists`, asserted precisely the behaviour being deliberately broken. Nothing else depended on it — `allowed_domains_initial` is always empty at the one production call site.

Two pre-existing nits found and left for follow-up, both the same "self-consistent wrong answer" flavour this fixes: `build_net_policy` ignores `--observe-domains`, and the startup summary prints `Allowlist: on` even under `--allow-all-domains`.

## Verification

Tests cover all four states through `classify_connect` rather than by inspecting list contents, plus the emptied-file reload, the deleted/unreadable reload, and check/proxy parity. No network. Mutation-checked both halves: reverting the gate fails three named tests, reverting the missing-file refusal fails two; filters matched 29 and 5 tests, not zero.

`cargo fmt --check`, `cargo clippy --all-targets -- -D warnings`, `--lib` 986, `unit_tests` 340, `--bin cplt` 54, `e2e` 156, `e2e_projects` 54, `e2e_guards` 123, `e2e_git_hardening` 11, `integration` 68 — all green.

Found by an internal security review, credited to @randax. An advisory will follow once this is released.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -3906,6 +3910,13 @@ fn build_net_policy(resolved: &config::Resolved, agent: agent::Agent) -> proxy::
...
         Vec::new()
     };
+    // Same intent bit `start_proxy_if_enabled` computes and hands the live
+    // proxy (via `DomainPolicy::build`). Without it, `cplt check net` would
+    // report ALLOWED for every host under an enabled-but-empty allowlist that
+    // the proxy blocks — a self-consistent wrong answer to the one command a
+    // user runs to verify the setup.
+    let allowlist_active = !default_allowlist.is_empty()
+        || (!resolved.allow_all_domains && resolved.allowed_domains.is_some());
     let allowed_domains = if default_allowlist.is_empty() {
```
[the remark this answers is not in the document]
Comment (Starefossen) [reply src/main.rs:3920]: Good catch, and it is the same defect this PR exists to remove. Fixed.

`cplt check net` now makes the same fail-closed refusal the proxy makes at startup, using the shared `missing_allowlist_error`, so a missing or unreadable file produces one message and exit 1 on both paths instead of check reporting BLOCKED-ALLOWLIST for a state the live run never reaches.

The underlying reason it slipped through: `parse_lines_file` returning `None` for both missing and unreadable is deliberate and load-bearing on the reload path, where it is what preserves the last-good list. It only becomes wrong at build time, where there is no last-good to preserve.
